engine/src/main/java/org/camunda/bpm/engine/externaltask/FetchAndLockBuilder.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.externaltask;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Fetch And Lock Builder used to enable a Fluent API that exposes all parameters for fetch and Lock operation.
+ */
+public interface FetchAndLockBuilder extends ExternalTaskQueryTopicBuilder {
```

Comment:
❌ Why does this builder need to inherit from `ExternalTaskQueryTopicBuilder` and define all APIs again? I thought we could execute a method on `FetchAndLockBuilder` like `#subscribe` which returns the `ExternalTaskQueryTopicBuilder`. So we have two fluent builders for two steps. First, configuring the global configs applied for all topic subscriptions, and second, configuring the topic subscriptions.